test/controllers/diary_entry_controller_test.rb

```diff
@@ -393,15 +394,22 @@ def test_comment_spammy
   end
 
   def test_list_all
+    diary_entry = create(:diary_entry)
+    geo_entry = create(:diary_entry, :latitude => 51.50763, :longitude => -0.10781)
+    public_entry = create(:diary_entry, :user_id => users(:public_user).id)
+
     # Try a list of all diary entries
     get :list
-    check_diary_list :normal_user_entry_1, :normal_user_geo_entry, :public_user_entry_1
+    check_diary_list diary_entry, geo_entry, public_entry
   end
 
   def test_list_user
+    diary_entry = create(:diary_entry)
+    geo_entry = create(:diary_entry, :latitude => 51.50763, :longitude => -0.10781)
```

Comment:
So the problem here is that by only creating those two entries we are no longer testing that the right posts get selected because there are no non-matching posts in the database. Much the same applies to the other list tests.